app/helpers/dams_objects_helper.rb

```diff
@@ -540,32 +522,34 @@ def grabPDFFile(parameters={})
 	#
 	# @param componentIndex (Optional) The component's index.
 	# @return A string that is the component's file use (type/role) value. E.g., "image-service", "audio-service", etc.
-	# @author David T.
+	# @author David T., hweng
 	#---
 	def grabFileUse(parameters={})
-
-		p = {:componentIndex=>nil}.merge(parameters)
-		componentIndex = p[:componentIndex]
-
-		prefix = (componentIndex != nil) ? "component_#{componentIndex}_" : ''
-		fieldData = @document["#{prefix}files_tesim"]
+    fieldData = file_data(parameters)
 		result = nil
 
 		if fieldData != nil
 			fieldData.each do |datum|
 				files = JSON.parse(datum)
 				if files["use"].end_with?("-service")
 					result = files["use"]
-					if(result == "document-service")
+					if(result == "document-service" ||result == "video-service" ||result == "audio-service")
```

Comment:
These conditionals are repeated almost exactly in the grabServiceFile method. Is there a refactoring opportunity here?